collector/info_schema_innodb_sys_tablespaces.go

```diff
@@ -0,0 +1,87 @@
+// Scrape `information_schema.innodb_sys_tablespaces`.
+
+package collector
+
+import (
+	"database/sql"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const innodbTablespacesQuery = `
+	SELECT
+	    SPACE,
+	    NAME,
+	    ifnull(FILE_FORMAT, 'NONE') as FILE_FORMAT,
+	    ifnull(ROW_FORMAT, 'NONE') as ROW_FORMAT,
+	    ifnull(SPACE_TYPE, 'NONE') as SPACE_TYPE,
+	    FILE_SIZE,
+	    ALLOCATED_SIZE
+	  FROM information_schema.innodb_sys_tablespaces
+	`
+
+var (
+	infoSchemaInnodbTablesspaceInfoDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, informationSchema, "innodb_tablespace_space_info"),
+		"The Tablespace information and Space ID.",
+		[]string{"tablespace_name", "file_format", "row_format", "space_type"}, nil,
+	)
+	infoSchemaInnodbTablesspaceFileSizeDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, informationSchema, "innodb_tablespace_file_size_bytes"),
+		"The apparent size of the file, which represents the maximum size of the file, uncompressed.",
```

Comment:
In this case, the upstream docs seem fine, it's just that context is missing for this docstring. One wouldn't know that it's about tablespaces without also reading the metric name... but it's a minor wart and shouldn't block you.

:+1: in any case...